### DISCUSSION TITLE
Stitching ALOS PALSAR interferograms (single track/and cross-track)

### DISCUSSION THREAD

**Discussion by Hassan22113** [isce-framework/isce2#893] [General]
Greetings dear InSAR community,

I am new to working with InSAR and I have a question. If we pick several ALOS PALSAR scenes from several adjacent tracks (SLC's) and make interferograms for each pair using stripmapApp.py in ISCE, we end up having several interferograms but they are all seperate maps. Is it possible to stich them (scenes from the same path and or adjacent paths) together and have one broad inteferogram over a wide area? 
Your suggestions will be appreciated. Thanks

**Comment by fdelgadodelapuente** [comment]
You can easily stitch ALOS PALSAR L1.0 images. In your input file just add them followed by a comma, like in the following 


	<component name="reference">
		<property name="IMAGEFILE">['IMG-HH-ALPSRP219546480-H1.0__D','IMG-HH-ALPSRP219546490-H1.0__D']</property>
		<property name="LEADERFILE">['LED-ALPSRP219546480-H1.0__D','LED-ALPSRP219546490-H1.0__D']</property>
		
		<property name="OUTPUT">reference</property>
	</component>

**Reply by Hassan22113** [reply]
Thank you very much for your reply. I will definitely give it a shot. How about if we are using data which is from ALOS PALSAR and is already in Level 1.1 ?

Another issue I am facing with Level 1.1 data when I process using ISCE stripmapApp is that since it's not in the raw format (1.0), I couldn't generate .slc's (which are formed in "formslc" step). I run steps one by one and it fails in the early steps. For processing Level 1.1 ALOS data, do I need to make some changes in xml files so that ISCE can skim through my Level 1.1 data and extract SLCs so I can proceed further with processing? Or any other suggestion ?
I am new to InSAR processing,  and I will really appreciate your valuable feedback.

Kind regards..

**Reply by fdelgadodelapuente** [reply]
If you want to process L1.1 data, you need to set the sensor to ALOS_SLC. In L1.1 format you cannot do the FBD2FBS oversampling and FBS2FBD downsampling to match the slant range resolution between the different data sets (FBS 28 MHz, FBD 14 MHz). ASF Vertex has the complete ALOS PALSAR catalog in L1.0 file format for the Americas, and most of the rest of the world. 

You can only stitch different frames in Level 1.0 format.

**Reply by Hassan22113** [reply]
That's very helpful:) Thank you so much for your detailed response. 
Wish you the very best :)

**Reply by Hassan22113** [reply]
Good day honourable,

I used L1.0 ALOSPALSAR pair for interferometry however when I obtained slcs after running formslc step and displayed the output of slc, I found that the slc amplitude and phase plots are streched in vertical direction and compressed in horizontal direction making it appear like a narrow and long rectangle unlike the image scene size (which is also rectangular but more squarish rectangular). The data I am using is FBD RAW and probably I need to be running a preprocessing step of resampling from FBD to FBS in order for the resolutions to match. I was wondering if this resampling is something needed to be explicitly shown in stripmapApp.xml just like we specify that we want to do unwrapping marking it as true for example. How would be the right way to doing that ?
Thanks a lot for your answers and support in my begining InSAR journey.....

**Reply by keepfighting666** [reply]
> You can easily stitch ALOS PALSAR L1.0 images. In your input file just add them followed by a comma, like in the following
> 
> ```
> <component name="reference">
> 	<property name="IMAGEFILE">['IMG-HH-ALPSRP219546480-H1.0__D','IMG-HH-ALPSRP219546490-H1.0__D']</property>
> 	<property name="LEADERFILE">['LED-ALPSRP219546480-H1.0__D','LED-ALPSRP219546490-H1.0__D']</property>
> 	
> 	<property name="OUTPUT">reference</property>
> </component>
> ```

Hello! I would like to know if it's possible to stitch images from adjacent tracks when their acquisition times are different. Can this stitching method still be applied in such cases?

**Reply by Hassan2211345** [reply]
Hi, I think across-track it wont work. Only along track.

**Comment by fdelgadodelapuente** [comment]
You did not set up properly the number of range and azimuth looks. To do so you need to know the sampling parameters that control the range and azimuth resolutions (range bandwidth, PRF, along track velocity, look angle). They are specific of each satellite and each satellite has different acquisition modes and beams.  They are complied in this document

https://drive.google.com/file/d/1UOtl3aGLAsO9Nunw5k7dPFImKaJSbnBj/view?usp=share_link
For ALOS the FBD and FBS range sampling are described in detail here
https://topex.ucsd.edu/sandwell/publications/117.pdf

**Reply by Hassan22113** [reply]
Thanks you very much for the upload of documents. I highly appreciate your response.  I will follow these and then setup the number of range and azimuth looks. 
Have a great day :)

**Comment by fdelgadodelapuente** [comment]
The pixel ratio for FBD is 4, so you need to apply 4 range looks and 16
azimuth looks for pixels of ~80 m.
You need to down/oversample only to process FBD-FBS cross
interferograms, otherwise the pixels of one of the images won't match those
of the other image. If you want to apply split spectrum corrections to
FBD-FBS interferograms, it is better to use the common bandwidth of 14 MHz
instead of the 28 MHz.

El jue, 31 oct 2024 a las 21:15, Hassan2211345 ***@***.***>)
escribió:

> Honourable Francisco Delgado,
>
> Based on documents provided, I realized that since both of my ALOS PALSAR
> scenes are in FBD, I probably don't necessarily need to change them from
> FBD to FBS and I just need to include the number of looks in the input file
> i.e., 4 rng-8 az in the case of FBD.
>
> Additionally, if I have more ALOS PALSAR scenes (in my case, all are in
> FBD RAW), I can just include a comma and add more scenes to image and
> leader files and they will be stitched.
> In Short, I don't a resample flag (FBD to FBS) since all the ALOS PALSAR
> scenes I have are in FBD (hence with the same range bandwidth of 14 MHz),
> am I right on that ?
>
> Thank you very much again!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/isce-framework/isce2/discussions/893#discussioncomment-11117074>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFQHWFBMC3ZQAHNYGJSMWNLZ6LBZPAVCNFSM6AAAAABQFWF2ISVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMJRG4YDONA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Francisco Delgado

**Reply by Hassan22113** [reply]
Thank you very much. I tried different range-azimuth looks and this time, I tried with 4-16 looks as you suggested, and I still get slc resolution not matchable to the frames I am using. Attached are my input files and a screenshot of my generated slc output. Seems like stitching has worked but resolution is still not satisfactory. Any further suggestion ? I run steps from startup-->formslc.
Thanks again !
![20241102_081829](https://github.com/user-attachments/assets/8ae5b09c-ddc2-4783-9fa0-a42c9998ed7b)
```

def configure_inputs(outDir): 
    """Write Configuration files for ISCE2 stripmapApp"""
    cmd_reference_config = '''<component>
    <property name="IMAGEFILE">
       <value>['IMG-HH-ALPSRP144766245-H1.0__A','IMG-HH-ALPSRP144766255-H1.0__A']</value>
    </property>
    <property name="LEADERFILE">
        <value>['LED-ALPSRP144766245-H1.0__A','LED-ALPSRP144766255-H1.0__A']</value>
    </property>
    <property name="RESAMPLE_FLAG">dual2single</property>
    <property name="OUTPUT">
        <value>P349I250</value>
    </property>
</component>'''

    print("writing reference.xml")
    with open(os.path.join(outDir, "reference.xml"), "w") as fid:
        fid.write(cmd_reference_config)
    
    cmd_secondary_config = '''<component>
    <property name="IMAGEFILE">
        <value>['IMG-HH-ALPSRP185026245-H1.0__A','IMG-HH-ALPSRP185026255-H1.0__A']</value>
    </property>
    <property name="LEADERFILE">
        <value>['LED-ALPSRP185026245-H1.0__A','LED-ALPSRP185026255-H1.0__A']</value>
    </property>
    <property name="RESAMPLE_FLAG">dual2single</property>
    <property name="OUTPUT">
        <value>P349C250</value>
    </property>
</component>'''
    
    print("writing secondary.xml")
    with open(os.path.join(outDir, "secondary.xml"), "w") as fid:
        fid.write(cmd_secondary_config)
    
    cmd_stripmap_config = '''<?xml version="1.0" encoding="UTF-8"?>
<stripmapApp>
  <component name="insar">
    <property name="sensor name">ALOS</property>
    <property name="demFilename">
        <value>/Users/home/aleemha/InSAR_analysis_hassan//P_349_250/data/output_NASADEM.tif</value>
    </property>
    <property name="range looks">4</property>
    <property name="azimuth looks">16</property>
  <component name="reference">
        <catalog>reference.xml</catalog>
    </component>
  <component name="secondary">
        <catalog>secondary.xml</catalog>
    </component>
    <property name="filter strength">0.3</property>
    <property name="do unwrap">True</property>
    <property name="unwrapper name">icu</property>
    <property name="do split spectrum">False</property>
    <property name="do dispersive">False</property>
</component>
</stripmapApp>'''

    print("writing stripmapApp.xml")
    with open(os.path.join(outDir, "stripmapApp.xml"), "w") as fid:
        fid.write(cmd_stripmap_config)

    print("XML files have been written to:", outDir)

# Main execution
outDir = '/Users/home/aleemha/InSAR_analysis_hassan/P_349_250'
os.makedirs(outDir, exist_ok=True)  # Ensure directory exists
configure_inputs(outDir)
```

**Reply by Hassan2211345** [reply]
And if I don't run the dual2single flag, and just use 4-16 range and azimuth looks, then my result look like this, which is better I think, but my concern is, why it is not the same size as the frame that we see when we download these scenes.
![Figure_1part2](https://github.com/user-attachments/assets/05e5dfdf-9051-4db1-8bd4-3489bbe28742)
